src/pmacfiltercontrol/pmacFilterControlWrapper.py

```diff
@@ -545,13 +547,18 @@ async def _start_singleshot(self, _) -> None:
     @_if_connected
     async def _set_shutter(self, shutter_state: int) -> None:
         if shutter_state == SHUTTER_CLOSED:
-            pos = self.shutter_pos_closed.get()
+            shutter_close = json.dumps({"command": "shutter_close"})
+            self._send_message(codecs.encode(shutter_close, "utf-8"))
+
         else:
             pos = self.shutter_pos_open.get()
-        await caput(f"{self.motors}:SHUTTER:POS", pos)
+            await caput(f"{self.motors}:SHUTTER:POS", pos)
```

Comment:
It would be simpler to just keep using the PVs for opening and closing manually and let the C app only have to worry about closing the shutter automatically when high3 is exceeded. Then you don't need the shutter close command or this logic branch here.